disttask/importinto/scheduler.go

```diff
@@ -103,31 +111,37 @@ func (s *importStepExecutor) RunSubtask(ctx context.Context, subtask *proto.Subt
 	}
 	s.logger.Info("split and run subtask", zap.Int32("engine-id", subtaskMeta.ID))
 
-	dataEngine, err := s.tableImporter.OpenDataEngine(ctx, subtaskMeta.ID)
-	if err != nil {
-		return err
-	}
-	// Unlike in Lightning, we start an index engine for each subtask, whereas previously there was only a single index engine globally.
-	// This is because the scheduler currently does not have a post-processing mechanism.
-	// If we import the index in `cleanupSubtaskEnv`, the dispatcher will not wait for the import to complete.
-	// Multiple index engines may suffer performance degradation due to range overlap.
-	// These issues will be alleviated after we integrate s3 sorter.
-	// engineID = -1, -2, -3, ...
-	indexEngine, err := s.tableImporter.OpenIndexEngine(ctx, common.IndexEngineID-subtaskMeta.ID)
-	if err != nil {
-		return err
+	var dataEngine, indexEngine *backend.OpenedEngine
+	if s.tableImporter.IsLocalSort() {
+		dataEngine, err = s.tableImporter.OpenDataEngine(ctx, subtaskMeta.ID)
+		if err != nil {
+			return err
+		}
+		// Unlike in Lightning, we start an index engine for each subtask,
+		// whereas previously there was only a single index engine globally.
+		// This is because the scheduler currently does not have a post-processing mechanism.
+		// If we import the index in `cleanupSubtaskEnv`, the dispatcher will not wait for the import to complete.
+		// Multiple index engines may suffer performance degradation due to range overlap.
+		// These issues will be alleviated after we integrate s3 sorter.
+		// engineID = -1, -2, -3, ...
```

Comment:
what is this comment used for “// engineID = -1, -2, -3, ...”？

Reply:
each subtask will have a index engine, that's how we alloc it's id